### ISSUE TITLE
[Feature Suggestions] A list of feature suggestions

### ISSUE THREAD

**Issue by TomGrobbe** [TomGrobbe/vMenu#115]
# vMenu Suggestions
Here you can request features for future vMenu versions. Please note that I'll only add features if I feel like they fit vMenu. If I don't add your feature, then you can add this yourself by forking the resource. However, I will not provide support if you decide to edit vMenu.

**Please read the list below before requesting a feature to make sure it has not been requested before.**

## Current WIP features:
- none, too busy with random things atm.

## Planned or requested features:
_This does not imply that i'll definitely make them, they might be planned but not at the top of the todo-list, or they may be suggested but won't be added, just listed here because people keep asking the same questions_
- Add ban message config option.
- Make multiple options for engine always on (vehicle type specific)
- Saving last used timecycle modifier list index.
- Add some object attachment menu. (probably won't happen, **decent** pr's welcome.)
- Add option in-game to disable controller support for the menu (user preference). (by @blockba5her)
- > Make some hooks available to users so they can achieve certain things when menu opens, etc. (by @schwim0341)
- Add some events like when weather change is started and when it ends.
- > Requesting the ability to spawn Bodyguards. With the option to choose which character model the bodyguard uses. Such as input name or just select from a list of bodyguards. no one seems to have made a server side bodyguard spawning menu for FiveM so i would really like this. (by @ReicitoE)

  Probably won't be added but it's still suggested.

## Completed requests
(not all added features are listed here, this only contains _some_ requests because i can't possibly list all of them, and i usually forget who originally suggested something since i get a shit ton of duplicate suggestions all the time)
- [✔ added v1.0.2](https://github.com/TomGrobbe/vMenu/releases/tag/v1.0.2) Ability to change the keybinding for keyboard/mouse.
- [✔ added v1.0.2](https://github.com/TomGrobbe/vMenu/releases/tag/v1.0.2) Ability to disable weather/time sync completely to let other resources handle it.
- [✔ added v1.0.4](https://github.com/TomGrobbe/vMenu/releases/tag/v1.0.4) a "noclip" option.
- [✔ added v1.0.8](https://github.com/TomGrobbe/vMenu/releases/tag/v1.0.8) Configurable list of Addon vehicles & peds to spawn.
- [✔ added in v1.1.0](https://github.com/TomGrobbe/vMenu/releases/tag/v1.1.0) Improve weapons menu (weapon categories etc...)
- [✔ Added in v1.1.1](https://github.com/TomGrobbe/vMenu/releases/tag/v1.1.1) Add camera (angle/rotation locking) options.
- [✔ Added in v1.1.1](https://github.com/TomGrobbe/vMenu/releases/tag/v1.1.1) (will be added soon) Improve weapon permissions support.
- [✔ Added in v1.3.0](https://github.com/TomGrobbe/vMenu/releases/tag/v1.3.0) Improve weather customization options, and possibly also add more time customization options.
- [✔ Added in v1.3.0](https://github.com/TomGrobbe/vMenu/releases/tag/v1.3.0)
  - refactor settings/addons being sent to the client (instead of all separate, do it in 1 single json string with only 1 event to speed things up).
  - move config options from __resource.lua to a new proper config system.
  - better configuration file with more config options
- [✔ Added in v1.3.0](https://github.com/TomGrobbe/vMenu/releases/tag/v1.3.0) additional indicator somewhere on the screen that shows up whenever you are talking through voice chat
- [✔ Added in v1.3.0](https://github.com/TomGrobbe/vMenu/releases/tag/v1.3.0) configurable list of pre-defined teleport locations
- [✔ Added in v1.3.0](https://github.com/TomGrobbe/vMenu/releases/tag/v1.3.0) configurable list of pre-defined blips on the map
- [✔ Added in v1.5.0](https://github.com/TomGrobbe/vMenu/releases/tag/v1.5.0) More driving styles for 'Drive to Waypoint' and 'Drive Around Randomly'

- ✔ Added some time around here:
  > adding in a way to go into the bans list in game click onto one of the bans and an option to change that bans duration or make it perm could be useful if u don't want to let a admin perm people but they temp ban a hacker that u need to full banned (by @skrubzilla)
- ✔ Added somewhere around here: MP character customization (freemode peds, like GTA:O)
- [✔ Added in v2.0.0](https://github.com/TomGrobbe/vMenu/releases/tag/v2.0.0) (I believe it was added in 2.0.0, might be earlier or later actually, not sure). Restore weapons on re-spawn
- [✔ Added in v2.2.0](https://github.com/TomGrobbe/vMenu/releases/tag/v2.2.0) Add the possibility to change the color of car headlights, this option would be extremely useful, thank you! :) Requested by @Takumi-Kanashii 
- [✔ Added in v2.2.0](https://github.com/TomGrobbe/vMenu/releases/tag/v2.2.0) (or some version around there can't remember) Weapon kits/load-outs saving and spawning options.
- [✔ Added in v2.2.1](https://github.com/TomGrobbe/vMenu/releases/tag/v2.2.1) Personal-vehicle options menu kind of like lambda menu, but mostly different.
- [✔ Added _after_ v2.2.2] Suggestion by @itismejoey "Add config option for setting a key (usually F3 in Lambda) to start/stop recording."

**Comment by Detoxz** [comment]
Possible Ban system through SQL instead of text file.
a Ban history/banned list as a button or option

**Comment by TomGrobbe** [comment]
@Detoxz Ban system through sql will probably never be added. Though what do you mean with ban history/banned list? There’s already a list of banned players available in the menu

**Comment by Detoxz** [comment]
The ability to view the list in game through a button via permissions  for players other than the server owner(if possible). 
Banhistory: the ability to check if the player has been banned before in the past and the reason stated on why and when via permissions
Bannedlist: in game button to view banned players via permissions

**Comment by Detoxz** [comment]
Another Suggestion:
Saved Locations to teleport to:
example: You stand in a location you like, and you press a button to save the location client side with a defined name similar to how peds and cars work. When you press the defined name, you teleport to the spot. Permissions support as well. It's somewhat of a home system / reference system

**Comment by 2dix** [comment]
Permission for each voice chat option

**Comment by justweb1** [comment]
Groups or Classes in Add-ons. For example:
```json
{
"vehicles": {
    "police": ["leo1", "leo2", "leo3"],
    "fire": ["fire1", "fire2"]
},
"peds": {
    "police": ["officer", "deputy", "trooper"],
    "civilian": ["bankrobber", "beachbum"]
},
"weapons": {
    "handgun": ["wristrocket"],
    "futuristic": ["laser"]
}}
```

**Comment by TomGrobbe** [comment]
@justweb1 That’s actually a great idea to do it using the existing addons.json file, had not thought of it like this before. I’ll add this in a future update 👍🏻

**Comment by datsun280zx** [comment]
@TomGrobbe When you do that, maybe add in where you can lock down certain groups like maybe make the vehicles LEO locked down to police only or Fire locked down to fire with add_principle group.police or group.fire

**Comment by TomGrobbe** [comment]
@datsun280zx Dynamic permissions like that are not possible atm. Would require a rewrite of the entire permissions system, which is 99% unlikely to ever happen.

**Comment by ACBobby** [comment]
I believe this is already on your to-do, but advanced MP ped customization? Facial features, etc. Still looking to use vMenu as a full replacement for our client side stuff and that's the only thing holding us back.

**Comment by skrubzilla** [comment]
adding in a way to go into the bans list in game click onto one of the bans and an option to change that bans duration or make it perm could be useful if u don't want to let a admin perm people but they temp ban a hacker that u need to full banned

**Comment by datsun280zx** [comment]
Keybind to teleport to waypoint? maybe f3?

**Comment by justweb1** [comment]
Maybe consider moving permissions to a json file kinda like the add-ons. That way this could be dynamically changed without having to reboot the server and also maybe dynamically in other scripts.

**Comment by TomGrobbe** [comment]
>Maybe consider moving permissions to a json file kinda like the add-ons. That way this could be dynamically changed without having to reboot the server and also maybe dynamically in other scripts.

Nope, that'd still require a resource reboot (just like it does right now) because of the way the menu's are created (only on initial resource load).
Besides, I'd rather keep it mostly to existing game features, rather than using external stuff.

**Comment by matty45** [comment]
- Add a way to spawned as saved ped skin
- Add a simple loadout system
- No scratches or damage decals on vehicles

**Comment by ACBobby** [comment]
Allow a configuration option to determine the style of menu-navigation. Arrow keys as it is now, or with the numpad like it is with Simple Trainer/Lambda.

**Comment by TomGrobbe** [comment]
@ACBobby numpad is impossible due to the fact that gta by default does not have all numpad keys mapped to controls, this would require every player to change their key layout. so this kind of defeats the purpose. If you're goint to make things unique like that where players need to rebind controls, you might as well edit the source directly for that as it's such a unique case.

**Comment by ACBobby** [comment]
Understood. Wasn't sure how much of an issue that would be so I do appreciate the detailed response as to why it isn't possible.

Are props on your to-do list at all?

**Comment by TomGrobbe** [comment]
@ACBobby like spawning objects and stuff? no. one of the fivem elements is working on a map editor that should be way better than anything i could implement myself.

**Comment by ACBobby** [comment]
Yup that was exactly what I was meaning. :P Any idea what the name of that editor is called? I'll check it out if there's any materials on it floating about. Definitely appreciate the timely responses this morning since these are essentially community questions I'm fielding at the moment.

**Comment by TomGrobbe** [comment]
Suggestion from the forums:
>Im just looking for a micrphone script, similar to arma exile, that lights up a mircrophone icon when you are speaking.
Many scripts offer the currently talking UI for everyone, but not just a simple mic icon that lights up for just your client.

**Comment by justweb1** [comment]
This should be pretty easy to implement.

> Suggestion from the forums:
> 
> > Im just looking for a micrphone script, similar to arma exile, that lights up a mircrophone icon when you are speaking.
> > Many scripts offer the currently talking UI for everyone, but not just a simple mic icon that lights up for just your client.

```NETWORK_IS_PLAYER_TALKING #
// 0x031E11F3D447647E 0xDA9FD9DB
// NetworkIsPlayerTalking
BOOL NETWORK_IS_PLAYER_TALKING(Player player);
returns true if someone is screaming or talking in a microphone```

**Comment by TomGrobbe** [comment]
@justweb1 This is _not_ a place where i ask stuff about how to do it.... it's a list of things for me to add in the future.

**Comment by schwim0341** [comment]
Would there be any hooks available in the resource that would allow me to implement something like this when the menu opens?

**Comment by MatsAnd** [comment]
Could you add a "Locations" section with the ability to have predefined locations from the addons config?

Example:
```json
"locations": {
    "Police Station": { "x": 10, "y": 540, "z": 124, "heading": 230 },
    "San Andreas": { "x": 250, "y": 180, "z": 74, "heading": 30 }
}
```

**Comment by MatsAnd** [comment]
How about adding a config file where we can specify default settings like voice proximity etc.?

**Comment by MatsAnd** [comment]
> And what should those predefined locations do? Blips? Teleport? Spawn ?
Teleport locations.

**Comment by matty45** [comment]
Customize your vehicles color using rgb values.

**Comment by TomGrobbe** [comment]
I've updated the first post in this issue. It should now be a lot easier to find existing requests.

**Comment by HumanTree92** [comment]
Disable opening vMenu on Controller. I have changed my vMenu key to F2 & i have had some people tell me that they can open vMenu on Controller.

**Comment by TomGrobbe** [comment]
@HumanTree92 impossible, unless they also use keyboard/mouse at the exact same time while also pressing keys on the controller.

**Comment by HumanTree92** [comment]
@TomGrobbe  Yeah it doesn't happen to me but others said it happened to them so that is probably what they where doing.

**Comment by HumanTree92** [comment]
@TomGrobbe Also would it be possible to say when the server starts that the weather is automatically set to say XMAS. Like a Config Option.

**Comment by TomGrobbe** [comment]
@HumanTree92 Yes, will be added in the future

**Comment by TomGrobbe** [comment]
@TomGrobbe “Add els compatibility convar”

**Comment by Drift91** [comment]
When the MP ped menu is done will it contain this hairstyle? https://i.imgur.com/bttDlmg.jpg
Every trainer I find, either client side or server side, doesn't have that hairstyle for some reason. Not sure if it's a technical problem or just a simple oversight. I'll be happier than Tom Cruise on that one episode of Oprah if you add it, even if it's not in the initial release of the MP menu.

Also I second HumanTree92's suggestion and I can't wait for the weapon loadout saving.

**Comment by TomGrobbe** [comment]
@Drift91 Not sure, I use gta’s known hairstyle id’s. If those include that one in the screenshot then yes it’ll be added, if not then I’m afraid it won’t be included.

**Comment by TomGrobbe** [comment]
>Edit:
>A couple of afterthoughts; Will the MP ped menu contain opacity for facial hair, eyebrows, etc? And will it contain tattoo customization? I finally gave in to my friend's demands to allow client scripts on my server so he can customize his tattoos. Would be able to ditch the client side trainer if tattoos are added to vMenu.


@Drift91 tattoos: probably not, unless new natives for that are found. There's no way (afaik) to remove tattoos from a player without completely removing all overlays. And there's also no way to _get_ current tattoos from a player. So, there would be no tattoo preview option and it's a "equip-and-you're-stuck-with-it" feature. Which makes this highly unlikely to be a thing.

**Comment by TomGrobbe** [comment]
@Drift91 You can request stuff here https://forum.fivem.net/c/general-discussion/feature-requests but i don't think it'll be added anytime soon if at all. There's probably natives already out there but those just haven't been documented/named yet. Or they use structs which is impossible to use in FiveM afaik.

**Comment by Pegi13** [comment]
I have an idea for player tags? Basically admins/moderators can give players tags. E.g. M>Online Players>[Player123]> Give tag > [Enter tag name] . When this action is completed the selected player will have a tag in chat just like JayJatAuto's. [The tag name you entered above] | [Player Name] | [the players message].

**Comment by Ultrunz** [comment]
Option for Neons to toggle on and off in the vehicle options

**Comment by TomGrobbe** [comment]
@Ultrunz That option already exists.

**Comment by Ultrunz** [comment]
My bad I never saw it. Thanks :)

**Comment by Ultrunz** [comment]
I meant more like an toggle all on and off thing. But that will do as it is :P. Love the transition script

**Comment by Samdvich** [comment]
**Drive to Waypoint
Drive Around Randomly**

Currently only the _“rushed”_ driving style is supported, can we get some choice over the driving style? 👍

**Comment by Samdvich** [comment]
**Tankers and Trailers**

An addition to change the livery of the Tankers and Trailers instead of spawning with random?

**Comment by TomGrobbe** [comment]
@Samdvich 🤔 that might be a little tricky.... i'll have to figure out how i'd actually implement that in a good way

**Comment by Samdvich** [comment]
@TomGrobbe perhaps an option to select the livery before spawning (in Vehicle Spawner > Trailers / Vehicle Spawner > Tankers)

**Comment by TomGrobbe** [comment]
Unlikely that it'll be done with that, as the vehicle spawner menu is completely dynamically generated at runtime.

**Comment by Samdvich** [comment]
@TomGrobbe
My bad, well, you're pretty good at what you do so you'll think of something pretty good no doubt.

**Comment by Pegi13** [comment]
Add the ability to attach vehicles. E.g. You can get trailers that can connect to more trailers however you need to attach them to each other via a menu like menyoo. Dont know how you would go about doing this but its an idea!

**Comment by Samdvich** [comment]
@pegi13 +1

**Comment by TomGrobbe** [comment]
> As mentioned on the FiveM forum, it would be nice to have a feature to either choose your own colors for the loaction display, or to have the street names another color than white. Possibly the heading as well.

**Comment by MFG4Ever** [comment]
My bad Tom, the street name post went into the wrong section. Next time I'll try using my eyes ;-)

**Comment by TheBaconSpace** [comment]
@TomGrobbe Hey Would you Be Able to Re Add  Weather From v1.0.2 ?

**Comment by TomGrobbe** [comment]
@TheBaconSpace uhm, what are you talking about exactly?

**Comment by TheBaconSpace** [comment]
@TomGrobbe Like 
![image](https://user-images.githubusercontent.com/16171045/47739833-a776b500-dc4c-11e8-81c4-4180286b40c6.png)

**Comment by TomGrobbe** [comment]
@TheBaconSpace i don't get what you're trying to request here. vMenu's weather options only got _new_ features, nothing was ever removed...

**Comment by TheBaconSpace** [comment]
@TomGrobbe  i remember in v1.2.1 There was a Weather Sub Menu But its not There any more

**Comment by TomGrobbe** [comment]
@TheBaconSpace Then you didn't configure things properly, it's definitely still there.

**Comment by TheBaconSpace** [comment]
@TomGrobbe I Know This isn't The Place But Would You Be Able To Make a V-Queue?

**Comment by Pegi13** [comment]
@TomGrobbe An option for Staff to cage players just like in Garrys Mod. Basically once you click cage the cage prop will spawn on the players pos and they will be caged until released.

**Comment by TomGrobbe** [comment]
@Pegi13 vMenu will never be a troll menu, not even for staff.

**Comment by HumanTree92** [comment]
@TomGrobbe I somewhat agree with @Pegi13 but the cage in Garry's Mod wasn't for trolling. It was to stop trolls instead of Freezing them. B/c sometimes if you freeze someone it could crash their game. But yes i agree that a cage player would be cool but not needed as you can freeze players instead but you take the chance of crashing the person you froze.

**Comment by TomGrobbe** [comment]
@HumanTree92 freezing someone doesn't crash their game in fivem. at least afaik nobody has ever reported that on discord or on the forums with a dump to provide info for it.

**Comment by HumanTree92** [comment]
@TomGrobbe yes I agree but afaik freezing someone in fivem doesn't crash their game either. I know I Gmod it happened alot.

**Comment by Pegi13** [comment]
This option i suggested is not intended for trolling, more or less stopping the trollers in their tracks.

**Comment by HumanTree92** [comment]
@Pegi13 Just use a freeze player command. EasyAdmin has it. Freeze them then talk to them. Or just kick/ban them.

**Comment by TomGrobbe** [comment]
@Pegi13 The only reasons why you'd want to put someone in a cage (that i can see), is for 1) being an asshole 2) trolling. Imo there's no valid reason why you'd want to do this for any other reason than 1 or 2. Either way, I'm not going to add this because i also feel like it'd be a waste of time for me to add this.

**Comment by inQntrol** [comment]
people forget that garrysmod and GTA are two entirely different games. caging people in gmod works because of the game and what you can and cant do. GTA works differently. it wont be the same. Caging people in gmod was always a shitty idea to begin with. if people are causing problems on your servers then why are you allowing them to stay on the server by caging them? just punish them the way everybody should by kicking them and giving that player slot to someone else or to the same person if they feel like they want another chance. there is a point for typing a reason as you're kicking someone.

**Comment by Samdvich** [comment]
Walking Styles And Facial Expressions For Peds

**Comment by TomGrobbe** [comment]
@Samdvich Facial expressions will be a part of the mp character creator, walking styles is already a thing

**Comment by Samdvich** [comment]
The walking Style only supports the MP Peds

**Comment by TomGrobbe** [comment]
@Samdvich That will stay like that because there’s no way to tell if a ped is male or female. So no way to tell which style to use

**Comment by Samdvich** [comment]
isn't _f_ usually female and _m_ usually male?

**Comment by TomGrobbe** [comment]
@Samdvich Not always a good way to check

**Comment by TomGrobbe** [comment]
@TomGrobbe Besides, if the player is spawned, there’s no way afaik to get the ped name, only the ped hash

**Comment by TomGrobbe** [comment]
That still requires looping through all of them. I’m not going to go through all that trouble and creating performance problems to add support for non mp peds. If  the proper mp character customization comes out I don’t see a reason why you wouldn’t want to use mp peds.

**Comment by Samdvich** [comment]
I would be fine, but I don't think the MP character is that versatile that you could recreate a homeless person or a construction worker or even someone normal like Chip. But, if you think so I guess I could make the swap and no longer run good old Chip as my character 😢

**Comment by Samdvich** [comment]
See equipped items; parachute, body armour, etc.

**Comment by Samdvich** [comment]
@TomGrobbe what are you quoting?

Seeing the item equipped on your ped, eg: wearing a parachute
(a parachute on your back when you have it equipped.)
http://cs2.gamemodding.net/images/8be09d56b74573398f842b501d703b9862cfc6bdb699540d0c9c92c31f44cb64.jpg

**Comment by AA341** [comment]
will be nice to add on option to spawn peds in game!

**Comment by Pegi13** [comment]
I see a lot of people having problems as a vehicle model name doesnt really tell what vehicle it actually is.
I have come up with an idea that involves changing the addons.json so that everytime you add an addon vehicle you can obviously put the model name then right beside put the real name of the vehicle <the real name is what players will see. E.g.
```
{
  "vehicles": [
     "748" : "747-800I"
  ],
  "peds": [
    "iman" : "Iron Man
  ],
  "weapons": [
    "",
    ""
  ]
}
```

**Comment by TomGrobbe** [comment]
@Pegi13 Vehicle names are already customizable using AddTextEntry though?

**Comment by Blizzard098** [comment]
Add the ability to enable radio (mostly for non-radio enabled cars such as police cars for those who use ELS)

**Comment by Tanel432** [comment]
Hi there.
Literally made an account to thank you for this and post one small feature request.
Would really like to see bit more options on the so-called autopilot feature. Like max speed limits. Driving styles and maybe more. Not sure what more can be done there. I am not that skilled stuff like this.
But I have seen at least these two have been made and working.
Thanks for advance.

**Comment by TomGrobbe** [comment]
@Tanel432 It's almost like you're watching over my shoulder right now 😏 https://www.vespura.com/hi/i/2018-11-17_19-53_68aa1_251.png

**Comment by Samdvich** [comment]
@TomGrobbe ah, so you're doing it- nice, what's the method?

**Comment by matt43121** [comment]
Could we get the weather & time function able to be used though rcon so we can change it when off the server?

**Comment by Tanel432** [comment]
@Samdvich Its the feature that allows you set cars to drive itself while you can go away or just too lazy to drive to somewhere.

**Comment by Samdvich** [comment]
@Tanel432 I know

**Comment by TomGrobbe** [comment]
@TomGrobbe Fireworks menu to place down and set off different types of fireworks like the interaction menu in gta:o

**Comment by JohnyMcFace** [comment]
Would it be possible to add an attachment option similar to Simple trainer? Attaching vehicles to other vehicles opens up a lot of opportunities

**Comment by ItsMat** [comment]
A few suggestions:
  - Toggle a no dirt option (When enabled vehicles don't get dirty).
  - Spawn vehicle clean.

**Comment by Agiger03** [comment]
Now we have recording options, but what about Director Mode? Would also help us a bit to make nice video clips.

**Comment by Drew101** [comment]
please allow for a snow option in permissions to include it in the dynamic weather system

**Comment by AwesomeBenny** [comment]
It would be nice to have some kind of MySQL support for permissions.

**Comment by TomGrobbe** [comment]
@AwesomeBenny mysql support for permissions? Can you explain what you're trying to suggest here? becaues it doesn't make much sense. Since permissions are all managed through ACL in FXServer.

**Comment by AwesomeBenny** [comment]
@TomGrobbe I was thinking on a MySQL table that you could register an identifier and set a specific group like admin. Instead of getting it from the permissions files, it would get it from a database and wouldn't require de server/resource to restart.

**Comment by TomGrobbe** [comment]
@AwesomeBenny it will always require a resource restart. Currently it doesn't require a server restart if you know what you're doing. (for exampe just execute the add_principal command in the server console and add a player to the group.admin principal, then tell them to (re)join the server and it'll work without a restart.

**Comment by AwesomeBenny** [comment]
@TomGrobbe doing the way that [I think] the essentialmode does wouldn't do the add_principal automatically when someone hops in the server it would give the person the correct permissions.

**Comment by AwesomeBenny** [comment]
@TomGrobbe I am suggesting that instead of add_principal for each user in the permissions.cfg, when someone enters on the server, it fetches the admin/mod rank from the database and adds that principal, or gives the permission set there.

**Comment by TomGrobbe** [comment]
> @TomGrobbe I am suggesting that instead of add_principal for each user in the permissions.cfg, when someone enters on the server, it fetches the admin/mod rank from the database and adds that principal, or gives the permission set there.

So basically, what you're saying is:
> Don't do x, but do y first and then do x anyway.

"Don't use add_principal for every user, but get the identifiers from a database and then use add_principal for every user"

**Comment by AwesomeBenny** [comment]
@TomGrobbe Not per say. I am only suggesting that you do X in another way. That is instead of getting it from a file, get the info from a database. Personaly if would feel easier to do stuff. It is just an opinion/suggestion tho.

**Comment by TomGrobbe** [comment]
> @TomGrobbe Not per say. I am only suggesting that you do X in another way. That is instead of getting it from a file, get the info from a database. Personaly if would feel easier to do stuff. It is just an opinion/suggestion tho.

Most people already struggle to setup one database for some RP framework, having another is probably going to cause a lot of trouble for a lot of people. And besides, vMenu doesn't read the permissions.cfg. It doesn't handle the permissions. That's all done by the server itself. vMenu just gets the permission from the server.

So, sorry, but there won't be a MySQL (or any other DB system for that matter) support for permissions. As that'd be extremely over-complicating things.

**Comment by IllusiveTea** [comment]
I also prefer the fact you dont need some database stuff just to edit the perms, overall having a database server for simple permissions isnt at all necessary and the current system works just as well (or even better, as you edit a single file quickly and easily).

**Comment by JohnyMcFace** [comment]
Any comment for the attachment suggestion? Just curious

**Comment by TomGrobbe** [comment]
@JohnyMcFace Unlikely to happen anytime soon if at all. But it’s noted 😉

**Comment by JohnyMcFace** [comment]
Better than nothing I guess. I was looking through simple traininer's code and couldn't find anything, but Im certainly not a developer.

**Comment by JohnyMcFace** [comment]
It could also be worthwhile to look in to how towing scripts work. Maybe having it based off that, and then changing x/y/z positioning 
Honestly not sure how simple trainer works it's magic

**Comment by TomGrobbe** [comment]
> It could also be worthwhile to look in to how towing scripts work. Maybe having it based off that, and then changing x/y/z positioning
> Honestly not sure how simple trainer works it's magic

@JohnyMcFace speaking of tow scripts, here you go:
https://forum.fivem.net/t/vtow-a-simple-vehicle-tow-script-including-trailers-support/191640?u=vespura

**Comment by JohnyMcFace** [comment]
Oooh yes I think I saw this the other day. Let me shoot it over and see what I find.

**Comment by TomGrobbe** [comment]
@JohnyMcFace Impossible, i posted it 5 minutes ago lol.

**Comment by JohnyMcFace** [comment]
Either A) Im on crack or B) I saw a similar post with a different mod

**Comment by TomGrobbe** [comment]
@JohnyMcFace I hope (for your sake) that it's option b 😅

**Comment by JohnyMcFace** [comment]
Me too lol. For oddly shaped trailers, such as... this https://www.gta5-mods.com/vehicles/semi-transport-hauler-add-on
That script wont work/will be very misplaced, correct?

**Comment by JohnyMcFace** [comment]
Actually, bad example. Thinking of like, the standard car hauler, where the platform isn't flat.

**Comment by TomGrobbe** [comment]
@JohnyMcFace Let's continue that discussion on that topic please.

**Comment by TomGrobbe** [comment]
> A few suggestions:
> 
> * Toggle a no dirt option (When enabled vehicles don't get dirty).
> * Spawn vehicle clean.

for the record, added this in the latest dev branch builds.

**Comment by TomGrobbe** [comment]
@Drift91 
> When the MP ped menu is done will it contain this hairstyle? https://i.imgur.com/bttDlmg.jpg
> Every trainer I find, either client side or server side, doesn't have that hairstyle for some reason. Not sure if it's a technical problem or just a simple oversight. I'll be happier than Tom Cruise on that one episode of Oprah if you add it, even if it's not in the initial release of the MP menu.
> 
> Also I second HumanTree92's suggestion and I can't wait for the weapon loadout saving.
> 
> **Edit:**
> A couple of afterthoughts; Will the MP ped menu contain opacity for facial hair, eyebrows, etc? And will it contain tattoo customization? I finally gave in to my friend's demands to allow client scripts on my server so he can customize his tattoos. Would be able to ditch the client side trainer if tattoos are added to vMenu.

I'm working on it again, is this what you meant with those specific overlays for hair styles?
https://www.vespura.com/hi/i/2018-12-01_01-05_ca3c8_316.png

They're not available for all hair styles because I couldn't find a list containing all of them, but I've found a list containing the first 22 hair styles with overlays.

Also:
- Opacity for facial hair etc: yes
- Tattoos: no. At least not until there's some way of properly supporting this. There are currently too many limiting factors with tattoos that I have decided to stop wasting my time researching them, it is taking way too long and I'm not making any progress on them. At least until everything else is done and released.

**Comment by Pegi13** [comment]
A speed limiter option located in "Vehicle Options". Basically if you set the speed limiter to 50MPH  your vehicle will not go over 50MPH and so on.

**Comment by HumanTree92** [comment]
Not sure if this is a bug or what but when you go to the components for a vehicle and if your vehicle is damaged you can toggle an extra on/off and it will repair the vehicle. Need to add an option to disable the repair of a vehicle if toggling an extra in the components. Or just disable it completely.

**Comment by TomGrobbe** [comment]
This is default gta behavior, and I do not intend to change that. You can get another resource to disable this.

**Comment by Pegi13** [comment]
Add another option to the speed limiter that allows you to input a max speed.

**Comment by TomGrobbe** [comment]
@Pegi13 @ToastinYou feel like doing another PR?

**Comment by ToastinYou** [comment]
For now, the speed limiter works, so let's just stick with it unless you, I, or someone else finds some time to implement it.

**Comment by ReicitoE** [comment]
Requesting the ability to spawn Bodyguards. With the option to choose which character model the bodyguard uses. Such as input name or just select from a list of bodyguards. no one seems to have made a server side bodyguard spawning menu for FiveM so i would really like this.

**Comment by Takumi-Kanashii** [comment]
> # vMenu Suggestions
> Here you can request features for future vMenu versions. Please note that I'll only add features if I feel like they fit vMenu. If I don't add your feature, then you can add this yourself by forking the resource. However, I will not provide support if you decide to edit vMenu.
> 
> **Please read the list below before requesting a feature to make sure it has not been requested before.**
> 
> ## Current WIP features:
> * MP character customization (freemode peds, like GTA:O)
> 
> ## Planned / requested features:
> * **Addon models**
>   
>   * Add categories into the addons.json file to create sub-menus for each category.
> * **Vehicles**
>   
>   * RGB vehicle color options
>   * Personal-vehicle options menu kind of like lambda menu, but mostly different
> * **Weapons**
>   
>   * Weapon kits/load-outs saving and spawning options.
>   * (Maybe: restore weapons on re-spawn)
> * **General features**
>   
>   * Global (chat) notifications for when someone gets kicked/banned/temp-banned (configurable).
>   * More staff actions like freezing players in the Online Players tab.
>   * _[UNLIKELY] MAYBE add MySQL support for bans._
> 
> ## Completed requests
> (not all added features are listed here, just the ones i didn't forget about)
> 
> * [✔ added v1.0.2](https://github.com/TomGrobbe/vMenu/releases/tag/v1.0.2) Ability to change the keybinding for keyboard/mouse.
> * [✔ added v1.0.2](https://github.com/TomGrobbe/vMenu/releases/tag/v1.0.2) Ability to disable weather/time sync completely to let other resources handle it.
> * [✔ added v1.0.4](https://github.com/TomGrobbe/vMenu/releases/tag/v1.0.4) a "noclip" option.
> * [✔ added v1.0.8](https://github.com/TomGrobbe/vMenu/releases/tag/v1.0.8) Configurable list of Addon vehicles & peds to spawn.
> * [✔ added in v1.1.0](https://github.com/TomGrobbe/vMenu/releases/tag/v1.1.0) Improve weapons menu (weapon categories etc...)
> * [✔ Added in v1.1.1](https://github.com/TomGrobbe/vMenu/releases/tag/v1.1.1) Add camera (angle/rotation locking) options.
> * [✔ Added in v1.1.1](https://github.com/TomGrobbe/vMenu/releases/tag/v1.1.1) (will be added soon) Improve weapon permissions support.
> * [✔ Added in v1.3.0](https://github.com/TomGrobbe/vMenu/releases/tag/v1.3.0) Improve weather customization options, and possibly also add more time customization options.
> * [✔ Added in v1.3.0](https://github.com/TomGrobbe/vMenu/releases/tag/v1.3.0)
>   
>   * refactor settings/addons being sent to the client (instead of all separate, do it in 1 single json string with only 1 event to speed things up).
>   * move config options from __resource.lua to a new proper config system.
>   * better configuration file with more config options
> * [✔ Added in v1.3.0](https://github.com/TomGrobbe/vMenu/releases/tag/v1.3.0) additional indicator somewhere on the screen that shows up whenever you are talking through voice chat
> * [✔ Added in v1.3.0](https://github.com/TomGrobbe/vMenu/releases/tag/v1.3.0) configurable list of pre-defined teleport locations
> * [✔ Added in v1.3.0](https://github.com/TomGrobbe/vMenu/releases/tag/v1.3.0) configurable list of pre-defined blips on the map
> * [✔ Added in v1.5.0](https://github.com/TomGrobbe/vMenu/releases/tag/v1.5.0) More driving styles for 'Drive to Waypoint' and 'Drive Around Randomly'

Hi, I would like you to add the possibility to change the color of car headlights, this option would be extremely useful, thank you! :)

**Comment by TomGrobbe** [comment]
@Takumi-Kanashii As far as I know, that's not possible. Unless there's some undocumented native for it.

**Comment by Takumi-Kanashii** [comment]
@TomGrobbe With the latest update of GTA online we can have colorful headlight, If we manage to put the headlights xenon on a car why not the new headlights colored

**Comment by TomGrobbe** [comment]
Then that is indeed using an undocumented dlc native, which I have no clue right now how to find that one. (because it's not documented yet). Xenon headlights are completely different. They are a vehicle mod, completely unrelated to colored headlights.

**Comment by Takumi-Kanashii** [comment]
I understand, In any case if you find how to add this option, it would be with a huge pleasure!

**Comment by QuadrupleTurbo** [comment]
Removing sub-menus from sub-menus without editing source code, for example, I'd like to just have in my vehicle spawner, the addon vehicle menu, instead of having all the other categories of vehicles. Just to add one other thing, more like an issue, I can't remove the "Spawn Inside Vehicle" and "Replace Previous Vehicle" within the permissions file, it still shows even tho I've disabled it.

**Comment by QuadrupleTurbo** [comment]
To be able change colours and fonts for personalisation, maybe even configurable menu names.

**Comment by TomGrobbe** [comment]
@QuadrupleTurbo the lattter: not possible without a full rewrite of basically everything. and fonts won't ever be a thing because MenuAPI calculates based on a specific font and font size. Not possible to do that dynamically without having to hardcode offets for everything and making a real mess.

The first about removing submenus: not possible without a full recode of everything either. Can't make menus like that dynamic. They are already dynamic in the sense of permissions allowing you to enable or disable options, if something doesn't work: make sure you're doing it correctly, if you are 100% sure you've done it correctly, get in touch on discord in the support channel, or ask on the topic, or make a bug report if you're sure it's a bug. But most importantly: provide (enough) INFO.

**Comment by QuadrupleTurbo** [comment]
Well at least I'm learning more to what's possible and what's not, thanks for the reply.

**Comment by NickM0822** [comment]
Add the ability to set a message for the auto ban of cheaters (like what it would say on the ban message)

**Comment by TomGrobbe** [comment]
@NickM0822 see above (https://github.com/TomGrobbe/vMenu/commit/f34fb34327c34ae04ff709c52e958ac35eb4a4d6)

**Comment by nathanctech** [comment]
Add console /kick and /ban commands so they can be used in chat or via rcon.

**Comment by TomGrobbe** [comment]
That’s already a thing. Though it’s only available through the server console, because it’s useless to use commands while ingame if you have the menu right there.

**Comment by polarim** [comment]
Is it possible to have an option to disable some modifications for vehicles, like disable the options to add weapons/the big ramps or whatever on the new vehicles and old ones?

**Comment by TomGrobbe** [comment]
nope, that's not possible, because those ID's change dynamically per vehicle, id 8 might be 'bumpers' or something like that for one vehicle, where id 8 on another might be 'mounted weapons'. And the names also change per-vehicle so no it's not possible to have dynamic permissions for this.

**Comment by polarim** [comment]
would it be possible to disable the whole category as an option for server owners, like turn off the 'tank' customization, bumpers, etc for different vehicles or is that also impossible?

**Comment by TomGrobbe** [comment]
@polarim There's way too many vehicles for that, same reason why spawning vehicles is not restricted per vehicle, but instead per vehicle category. Adding more permissions for restricting the editing of vehicles shouldn't be needed because you can already prevent players from spawning vehicles that you don't want them to have.

**Comment by polarim** [comment]
@TomGrobbe ye, i was just curious because some of the newer vehicles are cool but their modifications are what ruin them for the servers, thank you though.

**Comment by KreyZ1320rp** [comment]
Downloaded the new version and it says resource can not be loaded, there is no __resource.lua in the file download, can this be the issue?

**Comment by TomGrobbe** [comment]
@blockba5her why... controller keybind literally does _not_ interfere with anything, only thing it does is the same open method for gta:o's interaction menu...

**Comment by Tanel432** [comment]
Hi there. Tanel again. The one who asked autopilot.  Its great but I expected bit more options.
Came here to ask would it possible to add an option to mark players or one specific player as friendly? This might solve those accidental hits when you press R near someone to reload a weapon or hit the car with a crowbar. I might be wrong here too. Not really sure how to fix it. Wish someone could make a script that removes this feature.

**Comment by TomGrobbe** [comment]
>This might solve those accidental hits when you press R near someone to reload a weapon or hit the car with a crowbar. I might be wrong here too. Not really sure how to fix it.

That's actually related to team damage/friendly fire. Not something that 'marking as a friend (how would that even work? marking their blip as friendly?)' would solve.

>Hi there. Tanel again. The one who asked autopilot. Its great but I expected bit more options.

It was already planned to expand this before you suggested it, whenever the very first auto pilot features were implemented. However it's still a working progress and more features will be added in the future. Just not sure when.

What features are you missing exactly that you'd like to see added?

**Comment by TomGrobbe** [comment]
Speed limit can be set in the vehicle options menu. There is no way to task a ped to drive at a specific speed. Peds that can’t make a corner is just how gta handles those tasks, there’s nothing I can do about it. Avoiding traffic is one of the driving tasks iirc. If not, one will be added.

**Comment by itismejoey** [comment]
Unsure if this was added/requested above, but being able to attach vehicles to each other (ie: boats to trailers) would be very nice. Currently it's something that we need to use another resource for.

**Comment by Solmex72** [comment]
Could we get an object editor like the Menyoo Object Spooner? 

[Menyoo Link](https://www.gta5-mods.com/scripts/menyoo-pc-sp)
[Spooner How-To](https://forums.gta5-mods.com/topic/64/menyoo-object-spooner-tutorial)

It is an amazing tool that would allow for entire maps to be created on FiveM by average players.

**Comment by blockba5her** [comment]
A config option for disabling the option to open the menu on controller. 

You can't modify the current from the back button and I know some people use that for PTT on Teamspeak/Discord so the menu opens when they try to talk.

Eg.
```
setr vmenu_enable_gamepad_open true # or false
```

**Comment by TomGrobbe** [comment]
@blockba5her This will be added, but not through a convar controllable by server owners. This will be a user preference option that people will be able to control themselves in the menu. The server owner should not be the one deciding this for the users.

**Comment by Yufai** [comment]
Within the Vehicle Spawner submenu, and the Saved Vehicle submenu, it would be nice to have it so that options that are un-accessible to be hidden, making the menu look cleaner.

**Comment by TomGrobbe** [comment]
@Yufai The reason I didn’t do this is because of list indexes. The current setup requires a specific index to be present.

**Comment by Yufai** [comment]
@TomGrobbe I see.

**Comment by AntwanR942** [comment]
Add a cool down for vehicle spawn. I want everyone to be able to spawn a vehicle without replacing the last but then players can just spam. This is where you could add a cooldown config within the permissions.cfg
Idea of what it would look like
###Vehicle Cooldown
Vehicle_Cooldown = true ###true if you want the cooldown, false if you dont
Vehicle_Cooldown_Timer = 60000 ###Time in milliseconds. 1 minute default

**Comment by itismejoey** [comment]
Add config option for setting a key (usually F3 in Lambda) to start/stop recording.

**Comment by TheGoodCook** [comment]
Hey, I saw that not all of the tattoo's are in the vMenu any plans on adding the remaining tattoo options?

**Comment by TomGrobbe** [comment]
@TheGoodCook which ones are missing?

**Comment by TheGoodCook** [comment]
There are a few like some face tattoo's the ones I am naming below are from the Wiki. Though can you also make the tattoos work for both male and female?

Head: ( Beach Bum Pack )
Pirate Skull
SURF LS
Little Fish
Shark
Tribal Butterfly
Val-de-Grace Logo
Money Rose

Body: ( Beach Bum )
Tribal Flower
Parrot
Greed is Good
Tribal Fish


Body: ( I'm not a Hipster )
Mustache
Pizza
Padlock
Oriental Mural
Peace
Origami
Snake Outline / Snake Shaded





Head: ( Business Update )
Cash is King
Bold Dollar Sign
Script Dollar Sign
$100

Body: ( Business Update )
$100 Bill
All Seeing Eye
Mermaid L.S.

**Comment by TomGrobbe** [comment]
@TheGoodCook What wiki? And I’m pretty sure most of these are already included. Though I’d need the actual overlay names instead of the label names. Can’t do anything with these.

**Comment by Custommuscle2** [comment]
Perhaps add the ability to remove bumpers/fenders alongside the remove doors option? not sure if this is even logistically possible, apart from the fact that these parts periodically fall of of cars in the game if the car is damaged enough. Would be a nice feature to have but nothing to drive oneself to madness over :)

**Comment by TomGrobbe** [comment]
@Custommuscle2 I don't think there's any (known) native for this atm. Might be able to find an undocumented one, but if I (or someone else) can't then this can't be added.

**Comment by Custommuscle2** [comment]
Totally understandable. Like I said, just would be a nice to have if it did exist lol. Thanks for the quick reply!

**Comment by TayUE4** [comment]
Ability to spawn Bodyguards/NPC. I haven't yet seen a server-sided function with the exception of Arma-III styled missions and PIS resources. I believe from what I've seen in the natives that it's completely possible, hopefully this feature can include setting the relationship of the spawned ped/model. This would be great for scenario control etc. Thanks in advance for consideration and the work you do with this resource.

**Comment by TomGrobbe** [comment]
That probably won't happen. But it's noted if I ever feel extremely bored and have a lot of free time ;)

**Comment by matt43121** [comment]
Having the ability to share personal vehicle key with others much like the ESX lock resource but no esx

**Comment by Tommo-101** [comment]
Possibly add some events or exports that can be controlled by other scripts to change the weather and time? 

Also one that could receive when the weather is changed so it sends the current weather. For example scripts could be then made that post weather updates to discord.

**Comment by TomGrobbe** [comment]
>Possibly add some events or exports that can be controlled by other scripts to change the weather and time?

That can already be done using either the existing events or using the `ExecuteCommand` native to execute the `vmenuserver <weather|time>` commands server side.

> Also one that could receive when the weather is changed so it sends the current weather. For example scripts could be then made that post weather updates to discord.

You can do this by just checking the current and previous weather type, but sure an event could be added. I'll do this later.

**Comment by Flohhhhh** [comment]
**The ability to save the last used timecycle modifier client side for persistence every time you play** (or even a server side setting determined by owner so everyone has the same? This would be useful for more atmospheric game modes like zombies.)

**Comment by Flohhhhh** [comment]
"Engine Always On" Should automatically not function on planes and helicopters, as it just causes buggy flashing lights.

**Comment by TomGrobbe** [comment]
add some way to change the default pre-filled ban reason text.

**Comment by itismejoey** [comment]
Since there is a now a development menu, could there be an option to teleport to a coord?

**Comment by TomGrobbe** [comment]
> Since there is a now a development menu, could there be an option to teleport to a coord?

Sure can ;) will do that tomorrow.

**Comment by digboy** [comment]
Vespura. You abandoned the IPL/nightclub integration? Didn't saw any news about it since months...

Anyway, you make an amazing job and are a exemplar developer <3. Thank you for everything.

**Comment by TomGrobbe** [comment]
It's not really abandoned, but just put on hold. Most likely it won't be ending up in vMenu, but as a separate resource instead. Because the performance is currently not good enough for it to be a feature in vMenu, causing roughly 2ms cpu time _on top_ of a normal vMenu install.

**Comment by rdata01** [comment]
Hello Tom, i would like to suggest adding a test menu (just like the screen effects one previous added on VMenu) to particles, listed on https://devtesting.pizza/particle-list/.

**Comment by TomGrobbe** [comment]
> Hello Tom, i would like to suggest adding a test menu (just like the screen effects one previous added on VMenu) to particles, listed on https://devtesting.pizza/particle-list/.

There's 2279 different particle effects... most don't work though. And im not planning on going through all of them. Some also cause fire, explosions and kill anyone near them. I'd be fine with making a small list of some 'fun' and harmless ones, but they'd just be there for fun, and it'll probably be a very small list.

**Comment by ToastinYou** [comment]
Idea from DOJRP Community Members: "No Hat" option under Vehicle Options which would remove their ped's hat from their head when they enter a vehicle and put it back on their ped when they exit the vehicle.

**Comment by TomGrobbe** [comment]
>Idea from DOJRP Community Members: "No Hat" option under Vehicle Options which would remove their ped's hat from their head when they enter a vehicle and put it back on their ped when they exit the vehicle.

That already happens by default in GTA. Some hats/props/helmets get removed, others stay... so im not quite sure what you're suggesting here?

**Comment by ToastinYou** [comment]
Gotcha.. so must be an issue with our addon peds. Thanks Vespura.

**Comment by ToastinYou** [comment]
Suggestion: Add options to rotate vehicles on their forward/reverse and right/left axis (to flip cars over in particular angles).

**Comment by MichaelBelgium** [comment]
Can there be an animation list in "player related options" ? Or is this already provided somewhere?

**Comment by TomGrobbe** [comment]
>Can there be an animation list in "player related options" ? Or is this already provided somewhere?

There are way too many 'animations' to list in a menu.

>There are 14.053 animation dictionaries.
>There are 138.133 animation clips.

**Comment by MichaelBelgium** [comment]
@TomGrobbe wow, okay - didn't know that. Then maybe an equivalent like the `addons.json` file? That u can provide which animations can be in the list?

**Comment by TomGrobbe** [comment]
> @TomGrobbe wow, okay - didn't know that. Then maybe an equivalent like the `addons.json` file? That u can provide which animations can be in the list?

Probably not, because then server owners will just blindly dump all animations in and will end up crashing clients.

**Comment by LySoon** [comment]
Custom Language config :?

**Comment by polarim** [comment]
is there a way to disable bulletproof tires being used?

**Comment by polarim** [comment]
is it possible to disable bullet proof tires through vmenu?

**Comment by TomGrobbe** [comment]
> Custom Language config :?

Nope, would require the entire menu to be recoded.


> is it possible to disable bullet proof tires through vmenu?

Not that I know of nope. Mods are all dynamically generated and differ per vehicle.

**Comment by TomGrobbe** [comment]
Add driving style info for this specific flag: https://forum.fivem.net/t/vmenu-v3-0-3-a-server-sided-menu-including-almost-full-permissions-support/88868/3562?u=vespura

**Comment by itismejoey** [comment]
Can we get that to be the same notification (above the map) as the weather? https://gyazo.com/9c843b135ccf38211fd296f0f42d21c5

**Comment by EricG66** [comment]
Add possibility for changing the menu position on the screen via a config file or json

**Comment by itismejoey** [comment]
@EricG66 That is an option that the client user has to choose. I doubt Vespura has any plans to add this as a server config, as he believes the user should have the choice (as mentioned for other client related settings).

**Comment by TomGrobbe** [comment]
> Add possibility for changing the menu position on the screen via a config file or json

You can already go into the misc settings and select the left or right side of the screen. Alignment is done to the user's safezone size (pause menu > settings > display > safe zone size).

Position of the menu is user preference and should not be dictated by the server owner.

**Comment by Tommo-101** [comment]
Think it was mentioned before but a vehicle spawning cool down feature would be great. Enabling server owners to enable/disable and set the cool down timer period.

**Comment by neirbonhoj** [comment]
Hook/event for when a vehicle is spawned, passing along the user ID and vehicle

**Comment by La-Boba** [comment]
[Feature Suggestion] Permission option to disable reticle site. 
Add a option to disable reticle sites base on permission. 
And maybe add a option to increase kick (recoil) on weapons, also permission based.

**Comment by TomGrobbe** [comment]
@La-Boba I could add the no reticle option but it would not be permissions based, it’d be user preference. Im not going to make server owners force this on players through vMenu.

vMenu is made for players, not for server owners trying to dictate every players’ steps.

The recoil is not really something that’s suited for a menu like this.

**Comment by La-Boba** [comment]
@TomGrobbe Makes sense. Although unrelated on the latest version the pointer doesnt seem to work. Although it was said that it was implemented. Not sure if that's just me.

**Comment by TomGrobbe** [comment]
@La-Boba you mean finger pointing? if you have anything else that manages that such as lambda menu then yes it won't work. works fine on stock servers

**Comment by ShyPoner** [comment]
More animations like from the After Hours DLC like the club dancing

**Comment by 4cc3s** [comment]
Hello, can you help me with this issue? I have all files in vMenu and console says me this errors:

> Loaded vMenuServer.net, Version=3.0.0.0, Culture=neutral, PublicKeyToken=null into ScriptDomain_1721505112 
Exception loading assembly CitizenFX.Core.Server: System.IO.FileNotFoundException: Unable to find the specified file. 


Server stack trace: 
at System.Runtime.InteropServices.Marshal.ThrowExceptionForHR (System.Int32 errorCode) [0x0000a] in <b75790b9c16849d4aa8c31d6541d670a>:0 
at (wrapper cominterop) CitizenFX.Core.IScriptHost.OpenHostFile(string) 
at (wrapper cominterop-invoke) CitizenFX.Core.IScriptHost.OpenHostFile(string) 
at CitizenFX.Core.MonoScriptRuntime+WrapScriptHost.OpenHostFile (System.String fileName) [0x00000] in /src/code/client/clrcore/MonoScriptRuntime.cs:288 
at (wrapper remoting-invoke-with-check) CitizenFX.Core.MonoScriptRuntime+WrapScriptHost.OpenHostFile(string) 
at (wrapper xdomain-dispatch) CitizenFX.Core.MonoScriptRuntime+WrapScriptHost.OpenHostFile(object,byte[]&,byte[]&,string) 

Exception rethrown at [0]: 
at (wrapper xdomain-invoke) CitizenFX.Core.MonoScriptRuntime+WrapScriptHost.OpenHostFile(string) 
at CitizenFX.Core.InternalManager.LoadAssembly (System.String name) [0x00000] in /src/code/client/clrcore/InternalManager.cs:120 
System.IO.FileNotFoundException: Could not load file or assembly 'vMenuServer.net, Version=3.0.0.0, Culture=neutral, PublicKeyToken=null' or one of its dependencies 
File name: 'vMenuServer.net, Version=3.0.0.0, Culture=neutral, PublicKeyToken=null' 
at System.AppDomain.Load (System.String assemblyString, System.Security.Policy.Evidence assemblySecurity, System.Boolean refonly) [0x00016] in <b75790b9c16849d4aa8c31d6541d670a>:0 
at System.AppDomain.Load (System.String assemblyString) [0x00000] in <b75790b9c16849d4aa8c31d6541d670a>:0 
at (wrapper remoting-invoke-with-check) System.AppDomain.Load(string) 
at System.Reflection.Assembly.Load (System.String assemblyString) [0x00005] in <b75790b9c16849d4aa8c31d6541d670a>:0 
at System.UnitySerializationHolder.GetRealObject (System.Runtime.Serialization.StreamingContext context) [0x00135] in <b75790b9c16849d4aa8c31d6541d670a>:0 
at System.Runtime.Serialization.ObjectManager.ResolveObjectReference (System.Runtime.Serialization.ObjectHolder holder) [0x00010] in <b75790b9c16849d4aa8c31d6541d670a>:0 
at System.Runtime.Serialization.ObjectManager.DoFixups () [0x0007f] in <b75790b9c16849d4aa8c31d6541d670a>:0 
at System.Runtime.Serialization.Formatters.Binary.ObjectReader.Deserialize (System.Runtime.Remoting.Messaging.HeaderHandler handler, System.Runtime.Serialization.Formatters.Binary.__BinaryParser serParser, System.Boolean fCheck, System.Boolean isCrossAppDomain, System.Runtime.Remoting.Messaging.IMethodCallMessage methodCallMessage) [0x00077] in <b75790b9c16849d4aa8c31d6541d670a>:0 
at System.Runtime.Serialization.Formatters.Binary.BinaryFormatter.Deserialize (System.IO.Stream serializationStream, System.Runtime.Remoting.Messaging.HeaderHandler handler, System.Boolean fCheck, System.Boolean isCrossAppDomain, System.Runtime.Remoting.Messaging.IMethodCallMessage methodCallMessage) [0x000a2] in <b75790b9c16849d4aa8c31d6541d670a>:0 
at System.Runtime.Serialization.Formatters.Binary.BinaryFormatter.Deserialize (System.IO.Stream serializationStream, System.Runtime.Remoting.Messaging.HeaderHandler handler, System.Boolean fCheck, System.Runtime.Remoting.Messaging.IMethodCallMessage methodCallMessage) [0x00000] in <b75790b9c16849d4aa8c31d6541d670a>:0 
at System.Runtime.Serialization.Formatters.Binary.BinaryFormatter.Deserialize (System.IO.Stream serializationStream, System.Runtime.Remoting.Messaging.HeaderHandler handler, System.Boolean fCheck) [0x00000] in <b75790b9c16849d4aa8c31d6541d670a>:0 
at System.Runtime.Serialization.Formatters.Binary.BinaryFormatter.Deserialize (System.IO.Stream serializationStream, System.Runtime.Remoting.Messaging.HeaderHandler handler) [0x00000] in <b75790b9c16849d4aa8c31d6541d670a>:0 
at System.Runtime.Serialization.Formatters.Binary.BinaryFormatter.Deserialize (System.IO.Stream serializationStream) [0x00000] in <b75790b9c16849d4aa8c31d6541d670a>:0 
at System.Runtime.Remoting.RemotingServices.DeserializeCallData (System.Byte[] array) [0x0001c] in <b75790b9c16849d4aa8c31d6541d670a>:0 
at (wrapper xdomain-invoke) CitizenFX.Core.InternalManager.CreateAssembly(byte[],byte[]) 
at (wrapper remoting-invoke-with-check) CitizenFX.Core.InternalManager.CreateAssembly(byte[],byte[]) 
at CitizenFX.Core.MonoScriptRuntime.LoadFile (System.String scriptFile) [0x00068] in /src/code/client/clrcore/MonoScriptRuntime.cs:128 
Failed to load script vMenuServer.net.dll


What can i do with it?

**Comment by itismejoey** [comment]
> Hello, can you help me with this issue? I have all files in vMenu and console says me this errors:
> 
> > Loaded vMenuServer.net, Version=3.0.0.0, Culture=neutral, PublicKeyToken=null into ScriptDomain_1721505112
> > Exception loading assembly CitizenFX.Core.Server: System.IO.FileNotFoundException: Unable to find the specified file.
> > Server stack trace:
> > at System.Runtime.InteropServices.Marshal.ThrowExceptionForHR (System.Int32 errorCode) [0x0000a] in :0
> > at (wrapper cominterop) CitizenFX.Core.IScriptHost.OpenHostFile(string)
> > at (wrapper cominterop-invoke) CitizenFX.Core.IScriptHost.OpenHostFile(string)
> > at CitizenFX.Core.MonoScriptRuntime+WrapScriptHost.OpenHostFile (System.String fileName) [0x00000] in /src/code/client/clrcore/MonoScriptRuntime.cs:288
> > at (wrapper remoting-invoke-with-check) CitizenFX.Core.MonoScriptRuntime+WrapScriptHost.OpenHostFile(string)
> > at (wrapper xdomain-dispatch) CitizenFX.Core.MonoScriptRuntime+WrapScriptHost.OpenHostFile(object,byte[]&,byte[]&,string)
> > Exception rethrown at [0]:
> > at (wrapper xdomain-invoke) CitizenFX.Core.MonoScriptRuntime+WrapScriptHost.OpenHostFile(string)
> > at CitizenFX.Core.InternalManager.LoadAssembly (System.String name) [0x00000] in /src/code/client/clrcore/InternalManager.cs:120
> > System.IO.FileNotFoundException: Could not load file or assembly 'vMenuServer.net, Version=3.0.0.0, Culture=neutral, PublicKeyToken=null' or one of its dependencies
> > File name: 'vMenuServer.net, Version=3.0.0.0, Culture=neutral, PublicKeyToken=null'
> > at System.AppDomain.Load (System.String assemblyString, System.Security.Policy.Evidence assemblySecurity, System.Boolean refonly) [0x00016] in :0
> > at System.AppDomain.Load (System.String assemblyString) [0x00000] in :0
> > at (wrapper remoting-invoke-with-check) System.AppDomain.Load(string)
> > at System.Reflection.Assembly.Load (System.String assemblyString) [0x00005] in :0
> > at System.UnitySerializationHolder.GetRealObject (System.Runtime.Serialization.StreamingContext context) [0x00135] in :0
> > at System.Runtime.Serialization.ObjectManager.ResolveObjectReference (System.Runtime.Serialization.ObjectHolder holder) [0x00010] in :0
> > at System.Runtime.Serialization.ObjectManager.DoFixups () [0x0007f] in :0
> > at System.Runtime.Serialization.Formatters.Binary.ObjectReader.Deserialize (System.Runtime.Remoting.Messaging.HeaderHandler handler, System.Runtime.Serialization.Formatters.Binary.__BinaryParser serParser, System.Boolean fCheck, System.Boolean isCrossAppDomain, System.Runtime.Remoting.Messaging.IMethodCallMessage methodCallMessage) [0x00077] in :0
> > at System.Runtime.Serialization.Formatters.Binary.BinaryFormatter.Deserialize (System.IO.Stream serializationStream, System.Runtime.Remoting.Messaging.HeaderHandler handler, System.Boolean fCheck, System.Boolean isCrossAppDomain, System.Runtime.Remoting.Messaging.IMethodCallMessage methodCallMessage) [0x000a2] in :0
> > at System.Runtime.Serialization.Formatters.Binary.BinaryFormatter.Deserialize (System.IO.Stream serializationStream, System.Runtime.Remoting.Messaging.HeaderHandler handler, System.Boolean fCheck, System.Runtime.Remoting.Messaging.IMethodCallMessage methodCallMessage) [0x00000] in :0
> > at System.Runtime.Serialization.Formatters.Binary.BinaryFormatter.Deserialize (System.IO.Stream serializationStream, System.Runtime.Remoting.Messaging.HeaderHandler handler, System.Boolean fCheck) [0x00000] in :0
> > at System.Runtime.Serialization.Formatters.Binary.BinaryFormatter.Deserialize (System.IO.Stream serializationStream, System.Runtime.Remoting.Messaging.HeaderHandler handler) [0x00000] in :0
> > at System.Runtime.Serialization.Formatters.Binary.BinaryFormatter.Deserialize (System.IO.Stream serializationStream) [0x00000] in :0
> > at System.Runtime.Remoting.RemotingServices.DeserializeCallData (System.Byte[] array) [0x0001c] in :0
> > at (wrapper xdomain-invoke) CitizenFX.Core.InternalManager.CreateAssembly(byte[],byte[])
> > at (wrapper remoting-invoke-with-check) CitizenFX.Core.InternalManager.CreateAssembly(byte[],byte[])
> > at CitizenFX.Core.MonoScriptRuntime.LoadFile (System.String scriptFile) [0x00068] in /src/code/client/clrcore/MonoScriptRuntime.cs:128
> > Failed to load script vMenuServer.net.dll
> 
> What can i do with it?

This isn't the place for bug reports. You should also read the known issues for support on this.

**Comment by 4cc3s** [comment]
@itismejoey I dont know where to read it, im new at github

**Comment by itismejoey** [comment]
> @itismejoey I dont know where to read it, im new at github

Read the known issues on the [FiveM Forum post](https://forum.fivem.net/t/vmenu-v3-0-3-a-server-sided-menu-including-almost-full-permissions-support/88868). Also, stop posting here as it's for feature suggestions and not support. Reply to the thread on the forums or join the Discord.

**Comment by typoguy** [comment]
Add a separate permission node for the MP Ped Customization submenu, so that it can be specifically excluded

**Comment by TomGrobbe** [comment]
@typoguy Never. If you don’t want players to use _nice_ features, then don’t allow them to use the bad one either. Simple as that.

**Comment by Custommuscle2** [comment]
Could you add the ability to define custom vehicle classes for the spawn menu? Like being able to change where vanilla vehicles appear/put addon cars in the same lists? I'm on a server where we are doing a lot of vehicle additions and some of the classes like muscle are just too broad even in the vanilla list.

**Comment by TomGrobbe** [comment]
> Could you add the ability to define custom vehicle classes for the spawn menu? Like being able to change where vanilla vehicles appear/put addon cars in the same lists? I'm on a server where we are doing a lot of vehicle additions and some of the classes like muscle are just too broad even in the vanilla list.

I am not planning on adding this. It's currently ordered by _actual_ vehicle class id's. If there are addon cars you can make them have another class if you feel like they belong in another class. No manual sorting will be added. If you only want an addon vehicle spawner menu you can download MenuAPI or some other menu/ui API for FiveM and create your own menu for this quite easily.

**Comment by phxgg** [comment]
What I would love to see is a configurable identifier ban. My issue is the following:
A friend of mine owns a net cafe, in which all the PCs run FiveM on a one time bought GTA 5. This means that all players have the same license, xbl, live and IP identifiers, and only different steam identifiers (since they player on their own steam accounts). So I'd appreciate if you could add a boolean for each identifier (steam, license, xbl, live, discord, ip) in order that we can choose whether we would like to ban that identifier or not. Should take you less than 10 minutes.

**Comment by TomGrobbe** [comment]
>Should take you less than 10 minutes.

except i don't have 10 minutes to work on that right now. maybe in the future, but not anytime soon, as with any other suggestions, it'll be on the list, but not done anytime soon since i literally have no time to spend on vMenu at all.

**Comment by Tristian1975** [comment]
The vmenu needs a animal selection because our departments is trying to run k-9 unit we can do it with this menu plz make this change in the near future plz read this and plz you can do this is couple days.

**Comment by itismejoey** [comment]
> The vmenu needs a animal selection because our departments is trying to run k-9 unit we can do it with this menu plz make this change in the near future plz...……...

It is an option already, look in the permissions file to enable it.
`setr vmenu_enable_animals_spawn_menu`

**Comment by Wack1992** [comment]
Add a permission that allows the server owner to decide wether or not people should be allowed to spawn more than one car. 

# Default Disable Replace Previous for when the resource starts.
setr vmenu_default_ReplacePrevious false -- If true people can't spawn more than one car, if false people can spawn more than one car.

Something like this for the permissions.cfg

**Comment by TomGrobbe** [comment]
That's already an option @Wack1992

**Comment by Wack1992** [comment]
@TomGrobbe Oh shit, Sweet, thank you very much :)

**Comment by Zerise** [comment]
I noticed on the new arena cars, some had the tuning option to seperately change the rear wheels from the front ones, but apparently they didnt save, so they return to stock ones when you spawn the car again. I thought about implementing this tuning option into a car and since Im mostly doing it to use the car on FiveM, it would be nice if you could fix that in the next updates. Love the work, keep it up :)

**Comment by TomGrobbe** [comment]
> I noticed on the new arena cars, some had the tuning option to seperately change the rear wheels from the front ones, but apparently they didnt save, so they return to stock ones when you spawn the car again. I thought about implementing this tuning option into a car and since Im mostly doing it to use the car on FiveM, it would be nice if you could fix that in the next updates. Love the work, keep it up :)

Are you able to use those options in vMenu's vehicle mod menu? because if so, then they are just standard vehicle mods and they are already saved by default, if not i'll have to look into it. Got a list of some of the cars that have these special wheels because I've never seen it before.

**Comment by Zerise** [comment]
I was able to change them yeah, but when I tried to spawn a saved car, they would spawn in the car with stock wheels always.

**Comment by TomGrobbe** [comment]
I'm not sure what you're talking about. Do you have an example of a specific vehicle that has this option? As far as I know it's only possible on motorcycles.

**Comment by Dennis-NL** [comment]
Discord webhooks. with ban/kick logs and some other loggs

**Comment by TomGrobbe** [comment]
There are already event handlers for that kind of stuff.

**Comment by Tommo-101** [comment]
Client side trigger to enable / disable access to the menu. Useful when only wan't players to have access to the menu enabled when in safe zones / spawn areas etc and disabled when outside these areas.

**Comment by TomGrobbe** [comment]
Not going to be added unfortunately. This is a menu/trainer, it's designed to work all the time. Not only if certain criteria are met.

**Comment by Jhudri** [comment]
listen to players talk while spectating them

**Comment by TomGrobbe** [comment]
@Jhudri That’s already a thing in OneSync. Not sure about legacy servers.

**Comment by sambeano7** [comment]
Could you add a place to attach cars to trailers? Say if I want to haul vehicles on a flatbed trailer. Right now they just roll off and aren’t being “hauled” just sitting on them temporarily pretty much

**Comment by Tristian1975** [comment]
We need a thing that we need to attach stuff to things and trailers too
especially

**Comment by sambeano7** [comment]
That’s what I meant 🤦‍♂️

**Comment by Mayne34** [comment]
Would it be possible to add the No-Clip invisibility to a togglable ability. With the use of some Female MP Peds supplied thru vMenu, Emotes like "mechanic" "Pushups" "situps" cannot be used. With the No-Clipping feature from before, Those Peds would take advantage of the emote while No-Clipping to the desired location.

**Comment by ic3cak** [comment]
I would like to suggest the addition of an option to change the  **engine audio**  through the menu. Similar to the add-on ped selection feature, this would allow users to choose from a variety of engine audio options for FiveM servers
